Remove share permission check from the file upload

sharidas opened the pull request:
Remove share permission check from the file upload
js file.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs first be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.

Please set the following labels:

- Set label "3 - To review" for review or "2 - Development" in case the PR still has open tasks
- Set label "backport-request" if backport is needed
- Assignment: assign to self
- Milestone: set the same as the ticket this PR fixes, or "development" by default
- Reviewers: pick at least one
-->

## Description
<!--- Describe your changes in detail -->
Remove `sharePermission` checks from the file-upload.js file. We are no where using it and there is a problem in the code due to this. When we know its a file, we just have to check the fileinfo ( in the if condition ). The problem this PR tries to address is, when files are uploaded, the file information should be shown. Instead of this, the directory information was shown. 

This also fixes the cancel button problem in the conflict dialog. The cancel button clicked once cancels the conflict dialog.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Fixes https://github.com/owncloud/core/issues/35056
- Fixes https://github.com/owncloud/core/issues/35057

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Remove `sharePermission` checks from the file-upload.js file. This solves 2 problems:
1. The file which has conflict will show the file information in the conflict dialog popup and not the directory info.
2. Cancel button of the conflict dialog pop up will work with one click, instead of multiple click issue mentioned at https://github.com/owncloud/core/issues/35057.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
-  Kindly see the screen share 
![granularPermission](https://user-images.githubusercontent.com/3600427/56344792-ee953a00-61db-11e9-9fa1-1460ac814785.gif)


## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
<!-- In case of incomplete PR, please list the open tasks here -->
- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

sharidas committed:
Remove share permission check from the file upload

Remove share permission check from the file upload
js file.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

individual-it commented:
solves the problem for me, should be covered in acceptance tests

PVince81 commented:
@sharidas please research with git blame to see why these checks were added in the first place.

I remember a special case with share folders that needed to be covered, we need to make sure that your change does not break that.

sharidas commented:
> I remember a special case with share folders and conflict dialog that needed to be covered, we need to make sure that your change does not break that.

I did a quick test by sharing a folder from one oC user to another ( local share ), and my change did not break this. Here is the screen share for the quick test I did: 
![sharefolder](https://user-images.githubusercontent.com/3600427/56349979-5dc45b80-61e7-11e9-83bf-6f88c3122f77.gif)
I hope this is what you meant.
Also the git blame was pointing towards my commit and I didn't found any trace where the sharePermissions was used in the file.

PVince81 commented:
@sharidas you need to git blame further in the past. I believe Github has some buttons for this.
please find the original PR that instroduced the permissions check as it might contain more information about the test case, to confirm that what you tested matches. from what I remember it was a rather twisted setup...

sharidas commented:
I guess there is a confusion with respect to the PR description or heading. Looking at the history of file-upload.js file, when I try to get the state of this file before my changes were applied, the variable `sharePermission` is not available. This variable is introduced from my commit. I had removed only the if part from https://github.com/owncloud/core/commit/521722a5e7a2500108feb0324efa5d689bac74ab#diff-9112ab761e9b96bef644fc2c1f6e35c1R853 , that is https://github.com/owncloud/core/pull/34687/commits/b9879c3922f49a0cfae73c312aa6c6a546fdb287.  This is not to consider the autorename for the new granular permission task. I should have also removed `sharePermission` variable from the commit which was introduced in the commit https://github.com/owncloud/core/pull/34687/commits/521722a5e7a2500108feb0324efa5d689bac74ab

jvillafanez reviewed (approved):
Looks good, not tested

PVince81 reviewed (approved):
:+1: in regard of the clarification.

PVince81 commented:
@sharidas please backport to release-10.2.0 as it fixes a bug introduced by a recent change found by QA

sharidas commented:
Backport to release-10.2.0 branch https://github.com/owncloud/core/pull/35082

phil-davis commented:
The backport was merged into `release-10.2.0` and the release branch was merged into `stable10` so the backporting is done here.
